src/Application.cpp

```diff
@@ -448,6 +449,29 @@ void Application::loop(void)
   float aqi_10min = _sensor.airQualityIndex(ten_minutes_avg_pm2p5);
   setLEDColorForAQI(aqi_10min);
 
+  // Ensure the WIFI remains connected.
+  if (WiFi.status() != WL_CONNECTED) {
+    Serial.print(F("    ERROR - WiFi status is "));
+    Serial.print(WiFi.status());
+
+    if ((timestamp - _last_wifi_reconnect_time) < AIR_QUALITY_SENSOR_WIFI_RECONNECT_DELAY_SECONDS) {
+       Serial.print(F(", waiting to attempt reconnect.\n"));
+      return;
+    }
+
+     Serial.print(F(", attempting to reconnect."));
+    _last_wifi_reconnect_time = timestamp;
+
+    if (WiFi.reconnect()) {
+      Serial.print(F("    WiFi reconnected with IP address = "));
+      Serial.print(WiFi.localIP());
+      Serial.print(F("\n"));
+    } else {
+      Serial.println(F("    ERROR - failed to initiate WiFi reconnection."));
+    }
+    return;
```

Comment:
So this skips the rest of the loop even if the wifi reconnects. Should the `return` only be called if the wifi fails to reconnect?

Reply:
works for me